app/views/govuk_publishing_components/components/_radio.html.erb

```diff
@@ -40,9 +49,9 @@
 
     <% if heading.present? %>
       <% if is_page_heading %>
-        <%= tag.legend class: "govuk-fieldset__legend govuk-fieldset__legend--xl gem-c-title" do %>
-          <%= tag.span(heading_caption, class: "govuk-caption-xl") if heading_caption.present? %>
-          <%= tag.h1 heading, class: "gem-c-title__text" %>
+        <%= tag.legend class: legend_classes do %>
+          <%= tag.span(heading_caption, class: "govuk-caption-#{size}") if heading_caption.present? %>
+          <%= tag.h1 heading, class: "gem-c-title__text govuk-fieldset__heading" %>
```

Comment:
Not your change (mine in fact from two years ago, sorry), but inline with my previous comment, could this `gem-c-title__text` class be part of this component? You might have to copy the class across.

Since we've now switched to individual component assets for all our applications if the radio component were included without the title component, this part of it would be unstyled.

Reply:
I've added the CSS needed for this